mlr-org/gallery/optimization/2023-02-01-shadow-variable-search/index.qmd

```diff
@@ -0,0 +1,159 @@
+The difference is that the termination criterion is the selection of a shadow variable.
+Selecting a shadow variable means that the best improvement is achieved by adding a feature that is unrelated to the target variable.
+Consequently, the variables not yet selected are also correlated to the target variable only by chance.
+Therefore, only the previously selected features have a true influence on the target variable.
+
+Feature selection algorithms are implemented with the `r ref("mlr3fselect::FSelector")` class.
+A complete list of all fselectors is available on the [website](https://mlr-org.com/fselectors.html).
+We load the `r ref("mlr_fselectors_shadow_variable_search", "shadow variable search")` with the `r ref("mlr3fselect::fs()")` function.
+The algorithm has no parameters.
+
+```{r 2023-02-01-shadow-variable-search-002}
+#| message: false
+
+library(mlr3verse)
+
+fs("shadow_variable_search")
+```
+
+# Task and Learner
+
+The objective of the `r ref("mlr_tasks_pima", text = "Pima Indian Diabetes")` data set is to predict whether a person has diabetes or not.
+The data set includes 768 patients with 8 measurements.
+
+```{r 2023-02-01-shadow-variable-search-003}
+task = tsk("pima")
+```
+
+```{r 2023-02-01-shadow-variable-search-004}
+#| code-fold: true
+#| column: page
+#| fig-width: 14
+#| fig-height: 3
+#| fig.cap: "Distribution of the features in the Pima Indian Diabetes data set."
+#| warning: false
+
+library(ggplot2)
+library(data.table)
+
+task = tsk("pima")
+data = melt(as.data.table(task), id.vars = task$target_names, measure.vars = task$feature_names)
+
+ggplot(data, aes(x = value, fill = diabetes)) +
+  geom_density(alpha = 0.5) +
+  facet_wrap(~ variable, ncol = 8, scales = "free") +
+  scale_fill_viridis_d(end = 0.8) +
+  theme_minimal() +
+  theme(axis.title.x = element_blank())
+```
+
+The data set contains missing values.
+
+```{r 2023-02-01-shadow-variable-search-005}
+task$missings()
+```
+
+Support vector machines cannot handle missing values.
+We impute the missing values with the `r ref("mlr_pipeops_imputehist", "histogram imputation")` method.
+
+```{r 2023-02-01-shadow-variable-search-006}
+learner = po("imputehist") %>>% lrn("classif.svm", predict_type = "prob")
+```
+
+# Feature Selection
+
+We can now start the feature selection directly with the `r ref("fselect()")` function.
+In addition to the task, learner and feature selection algorithm, we have to select a `r ref("Resampling", text = "resampling strategy")` and `r ref("Measure", text = "performance measure")` to determine how the performance of a feature subset is evaluated.
+
+```{r 2023-02-01-shadow-variable-search-007}
+instance = fselect(
+  method = fs("shadow_variable_search"),
+  task = task,
+  learner = learner,
+  resampling = rsmp("cv", folds = 3),
+  measures = msr("classif.auc")
+)
+```
+
+The `r ref("fselect()")` function returns an `r ref("mlr3fselect::FSelectInstanceSingelCrit")` that includes an archive with all evaluated feature sets.
```

Comment:
Also ich finde es irgendwie verwirrend das archive hier zu zeigen. Entweder nicht zeigen oder ein bissl erklären. 
Also in der Algorithmus-Beschreibung oben, ist ja eine iteration: 
"Fitte ein Model für jedes neue Feature und füge das hinzu, das am meisten bringt". 
Die Darstellung im Archive ist ja viel detaillierter weil halt jedes Model dargestellt wird.

Reply:
Okay also zeige ich lieber den Optimization Path